File: DemoServer/AdditionalLanguages/python/queries/filteringResultsBasics/filteringResultsBasics.py
from typing import List
from demo_example import Example
from models import Employee

class FilteringResultsBasic(Example):
    def run(self, run_params=None) -> List[Employee]:
        #region Demo
        with self.document_store_holder.store().open_session() as session:
            #region Step_1
            filtered_query = session.query(object_type=Employee)
            #endregion
            #region Step_2
            filtered_query = filtered_query.where_equals("FirstName", "Anne")
            #endregion

            #region Step_3
            filtered_employees = list(filtered_query)
            #endregion
        #endregion

            return filtered_employees

File: DemoServer/AdditionalLanguages/php/queries/filteringResultsBasics/FilteringResultsBasics.php
<?php

namespace RavenDB\Demo\queries\filteringResultsBasics;

//region Usings
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Employee;

class FilteringResultsBasics
{
    public function __invoke(): array
    {
        //region Demo
        $filteredEmployees = [];

        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {
        //region Step_1
        $filteredQuery = $session->query(Employee::class)
        //endregion
                //region Step_2
                ->whereEquals("FirstName", "Anne");
                //endregion

            //region Step_3
            $filteredEmployees = $filteredQuery->toList();
            //endregion
            
        } finally {
            $session->close();
        }
        //endregion

        return $filteredEmployees;
    }
}
